File: map-raster-tile/MapRasterTiles/MainViewController.m
/*
 * Copyright (c) 2011-2018 HERE Europe B.V.
 * All rights reserved.
 */
#import "MainViewController.h"
#import "ImageTileSource.h"
#import "URLTileSource.h"
#import <NMAKit/NMAKit.h>

@interface MainViewController ()
@property (weak, nonatomic) IBOutlet NMAMapView* mapView;

@property (weak, nonatomic) IBOutlet UISegmentedControl* segmentedCtrl;
@end

@implementation MainViewController
{
    NMAMapTileLayer* _imageSourceTile;
    NMAMapTileLayer* _urlSourceTile;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Create an instance of URL raster tile source
    if (_urlSourceTile == nil)
    {
        _urlSourceTile = [URLTileSource new];
    }
    // Create an instance of Image raster tile source
    if (_imageSourceTile == nil)
    {
        _imageSourceTile = [ImageTileSource new];
    }
}

/**
 * Remove all tile layers which have been added to NMAMapView
 **/
- (void)removeTileLayers
{
    for (id layer in [_mapView mapTileLayers])
    {
        [_mapView removeMapTileLayer:layer];
    }
}

/**
 * Add NMAMapTileLayer when segmented control value changed.
 **/
- (IBAction)segmentedCtrlValueChanged:(id)sender
{
    switch (_segmentedCtrl.selectedSegmentIndex)
    {
    case 0:
        [self removeTileLayers];
        [_mapView addMapTileLayer:_urlSourceTile];
        break;
    case 1:
        [self removeTileLayers];
        [_mapView addMapTileLayer:_imageSourceTile];
        break;

    default:
        break;
    }
}

@end

File: map-raster-tile-swift/MapRasterTiles/MainViewController.swift
/*
 * Copyright (c) 2011-2018 HERE Europe B.V.
 * All rights reserved.
 */

import UIKit
import NMAKit

class MainViewController: UIViewController {

    @IBOutlet weak var segmentedCtrl: UISegmentedControl!
    @IBOutlet weak var mapView: NMAMapView!

    lazy var imageSourceTile: NMAMapTileLayer! = ImageTileSource()
    lazy var urlSourceTile: NMAMapTileLayer! = URLTileSource()

    /**
     * Remove all tile layers which have been added to NMAMapView
     **/
    private func removeTileLayers() {
        for layer in mapView.mapTileLayers() {
            mapView.remove(mapTileLayer: layer)
        }
    }

    /**
     * Add NMAMapTileLayer when segmented control value changed.
     **/
    @IBAction func segmentedCtrlValueChanged(_ sender: Any) {
        switch self.segmentedCtrl.selectedSegmentIndex {
        case 0:
            self.removeTileLayers()
            if let tile = self.urlSourceTile {
                self.mapView.add(mapTileLayer: tile);
            }
        case 1:
            self.removeTileLayers()
            if let tile = self.imageSourceTile {
                self.mapView.add(mapTileLayer: tile);
            }
        default:
            print("Trigger not implemented")
        }
    }
}
